avalon/tools/workfiles/app.py

```diff
@@ -295,11 +295,12 @@ def __init__(self):
 
         self._last_selected_task = None
 
-    def set_asset(self, asset_id):
+    def set_asset(self, asset):
 
-        if asset_id is None:
+        assets = []
+        if asset is not None:
             # Asset deselected
-            return
+            assets.append(asset)
```

Comment:
What exactly changed that required this check to be the opposite and have no return on `None`?
Is it required to run through to reset something else?

Would this be more readable as the following:
```python
assets = [asset] if asset else []
```

Also the `# Asset deselected` comment seems to now be wrong with the change of the condition too.

Reply:
Comment will be removed, good spot, and the condition is from future PR, can change it as you wrote...